include/util/guidance/name_announcements.hpp

```diff
@@ -64,51 +129,24 @@ inline bool requiresNameAnnounced(const std::string &from_name,
     const auto name_is_contained =
         boost::starts_with(from_name, to_name) || boost::starts_with(to_name, from_name);
 
-    const auto checkForPrefixOrSuffixChange = [](
-        const std::string &first, const std::string &second, const SuffixTable &suffix_table) {
+    const auto checkForPrefixOrSuffixChange =
+        [](const std::string &first, const std::string &second, const SuffixTable &suffix_table) {
+            std::string first_prefix, first_suffix, second_prefix, second_suffix;
+            std::tie(first_prefix, first_suffix, second_prefix, second_suffix) =
+                decompose(first, second);
 
-        const auto first_prefix_and_suffixes = getPrefixAndSuffix(first);
-        const auto second_prefix_and_suffixes = getPrefixAndSuffix(second);
+            const auto checkTable = [&](const std::string &str) {
+                // workaround for cucumber tests:
```

Comment:
:scream_cat:

Reply:
well... it's a bad side-effect of having all the different roads that are just two letters and include `N/E/S/W` in their lettering :(. In the end, this check prevents matching suffixes that are just 1 of two letters. This could be relevant in `1N` `1S`, if that should exist anywhere. In general, it is more helpful on cucumber tests, where we would need to ensure that no name is actually just two letters (which is mostly the case right now)